TTS/tts/models/tacotron2.py

```diff
@@ -28,35 +27,60 @@ def __init__(self,
                  bidirectional_decoder=False,
                  double_decoder_consistency=False,
                  ddc_r=None,
-                 gst=False):
+                 speaker_embedding_dim=None,
+                 gst=False,
+                 gst_embedding_dim=512,
+                 gst_num_heads=4,
+                 gst_style_tokens=10):
         super(Tacotron2,
               self).__init__(num_chars, num_speakers, r, postnet_output_dim,
                              decoder_output_dim, attn_type, attn_win,
                              attn_norm, prenet_type, prenet_dropout,
                              forward_attn, trans_agent, forward_attn_mask,
                              location_attn, attn_K, separate_stopnet,
                              bidirectional_decoder, double_decoder_consistency,
-                             ddc_r, gst)
-        decoder_in_features = 512 if num_speakers > 1 else 512
-        encoder_in_features = 512 if num_speakers > 1 else 512
-        proj_speaker_dim = 80 if num_speakers > 1 else 0
-        # base layers
+                             ddc_r, gst, gst_embedding_dim, gst_num_heads, gst_style_tokens)
+
```

Comment:
The same here maybe can share some stuff in the abstract class.

Reply:
I made the changes

Reply:
I didn't remember that!
I just sent the commit that implements these tests!